src/themes/from_zed_theme.rs

```diff
@@ -1,210 +1,287 @@
+        ("Atelier Heath Dark", ZED_THEME_ATELIER_URL),
+        ("Atelier Heath Light", ZED_THEME_ATELIER_URL),
+        ("Atelier Lakeside Dark", ZED_THEME_ATELIER_URL),
+        ("Atelier Lakeside Light", ZED_THEME_ATELIER_URL),
+        ("Atelier Plateau Dark", ZED_THEME_ATELIER_URL),
+        ("Atelier Plateau Light", ZED_THEME_ATELIER_URL),
+        ("Atelier Savanna Dark", ZED_THEME_ATELIER_URL),
+        ("Atelier Savanna Light", ZED_THEME_ATELIER_URL),
+        ("Atelier Seaside Dark", ZED_THEME_ATELIER_URL),
+        ("Atelier Seaside Light", ZED_THEME_ATELIER_URL),
+        ("Atelier Sulphurpool Dark", ZED_THEME_ATELIER_URL),
+        ("Atelier Sulphurpool Light", ZED_THEME_ATELIER_URL),
+        ("Ayu Dark", ZED_THEME_AYU_URL),
+        ("Ayu Light", ZED_THEME_AYU_URL),
+        ("Ayu Mirage", ZED_THEME_AYU_URL),
+        ("Catppuccin Frappé", ZED_THEME_CATPPUCCIN_URL),
+        ("Catppuccin Latte", ZED_THEME_CATPPUCCIN_URL),
+        ("Catppuccin Macchiato", ZED_THEME_CATPPUCCIN_URL),
+        ("Catppuccin Mocha", ZED_THEME_CATPPUCCIN_URL),
+        ("Gruvbox Dark Hard", ZED_THEME_GRUVBOX_URL),
+        ("Gruvbox Dark Soft", ZED_THEME_GRUVBOX_URL),
+        ("Gruvbox Dark", ZED_THEME_GRUVBOX_URL),
+        ("Gruvbox Light Hard", ZED_THEME_GRUVBOX_URL),
+        ("Gruvbox Light Soft", ZED_THEME_GRUVBOX_URL),
+        ("Gruvbox Light", ZED_THEME_GRUVBOX_URL),
+        ("Monokai", ZED_THEME_MONOKAI_URL),
+        ("Monokai-ST3", ZED_THEME_MONOKAI_ST3_URL),
+        ("One Dark", ZED_THEME_ONE_URL),
+        ("One Light", ZED_THEME_ONE_URL),
+        ("Rosé Pine Dawn", ZED_THEME_ROSE_PINE_URL),
+        ("Rosé Pine Moon", ZED_THEME_ROSE_PINE_URL),
+        ("Rosé Pine", ZED_THEME_ROSE_PINE_URL),
+        ("Sandcastle", ZED_THEME_SANDCASTLE_URL),
+        ("Solarized Dark", ZED_THEME_SOLARIZED_URL),
+        ("Solarized Light", ZED_THEME_SOLARIZED_URL),
+        ("Summercamp", ZED_THEME_SUMMERCAMP_URL),
+    ]
+    .iter()
+    .map(|(name, url)| ThemeDescriptor::ZedThemeURLMap(name, url))
+    .collect()
+}
+
+pub(crate) fn from_url(name: &'static str, url: &'static str) -> anyhow::Result<Theme> {
+    let path = std::path::PathBuf::from(url);
+    let file_name = path
+        .file_name()
+        .ok_or_else(|| anyhow::anyhow!("The url ({:?}) should contain a file name.", url))?
+        .to_string_lossy();
+    let json_str = cache_download(url, "zed-themes", &file_name)?;
+    let manifest: ZedThemeManiftest = serde_json5::from_str(&json_str).map_err(|error| {
+        anyhow::anyhow!("Cannot parse JSON downloaded from {url:?} due to:\n{error:#?}")
+    })?;
+
     Ok(manifest
         .themes
-        .into_iter()
-        .flat_map(|theme| -> anyhow::Result<Theme> {
+        .iter()
+        .find(|theme| theme.name == name)
+        .ok_or_else(|| anyhow::anyhow!("could not find theme '{}'", name))?
+        .clone()
+        .into())
+}
+
+impl From<ThemeContent> for Theme {
+    fn from(theme: ThemeContent) -> Self {
```

Comment:
I think we should avoid using `From`, because based on my experience, it makes tracking this part of the code difficult, IIRC running "Go to definition" on `into()` doesn't always lead to the specialized implementation.

We should keep this as a simple top-level function first, until one day we need a function to abstract over `T: From<ThemeContent>`.

Reply:
Also, unrelated to the above points, using a top-level function keeps the indentation of the body unchanged, thus greatly reducing space-wise diffs.